src/com/lilithsthrone/game/character/CharacterUtils.java

```diff
@@ -1363,6 +1364,15 @@ public static void equipClothing(GameCharacter character, boolean replaceUnsuita
 		Colour primaryColour = ColourListPresets.ALL.getPresetColourList().get(Util.random.nextInt(ColourListPresets.ALL.getPresetColourList().size())),
 				secondaryColour = ColourListPresets.ALL.getPresetColourList().get(Util.random.nextInt(ColourListPresets.ALL.getPresetColourList().size())),
 				lingerieColour = ColourListPresets.LINGERIE.getPresetColourList().get(Util.random.nextInt(ColourListPresets.LINGERIE.getPresetColourList().size()));
+
+		String clothingPattern = "none";
+	    int patternIndex = (int) (Math.random() * Pattern.getAllPatterns().values().size());
+	    for(Pattern pattern: Pattern.getAllPatterns().values()) if (--patternIndex < 0) {
```

Comment:
This code is terribly confusing; replace with Pattern pattern = Pattern.getAllPatterns()[patternIndex]; then your math.random check